zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -352,15 +351,12 @@ where
                             })?;
 
             let chain_info = match response {
-                ReadResponse::ChainInfo(Some(chain_info)) => chain_info,
+                ReadResponse::ChainInfo(chain_info) => chain_info,
```

Comment:
We should probably move `zip317::select_mempool_transactions(mempool).await?;` below the `ChainInfo` request in case the tip changes.

Reply:
I will do this as part of the long polling refactor, where we need to split up fetching data and generating the template anyway.